lib/wallet/src/Cardano/Wallet/Write/Tx.hs

```diff
@@ -543,3 +704,11 @@ withStandardCryptoConstraint
 withStandardCryptoConstraint era a = case era of
     RecentEraBabbage -> a
     RecentEraAlonzo -> a
+
+withCLIConstraint
+    :: RecentEra era
+    -> (CLI (ShelleyLedgerEra era) => a)
```

Comment:
Going with later. Requires `IsRecentEra era =>` rather than `RecentEra era` for us to easily get the constraint without pattern-matching or calling a helper. As [ADP-2353](https://input-output.atlassian.net/browse/ADP-2353) may affect the choice between passing `IsRecentEra era =>` or value-level `RecentEra` to functions, it seems better to see how that plays out first.